radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java

```diff
@@ -4,37 +4,44 @@
 import com.radixdlt.client.core.atoms.particles.Quark;
 import org.radix.serialization2.DsonOutput;
 import org.radix.serialization2.SerializerId2;
+import org.radix.time.Timestamps;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 
 /**
  * A quark that keeps a specific timestamp.
  * Currently used to keep atom-specific timestamps in TimestampParticle.
  */
 @SerializerId2("CHRONOQUARK")
 public final class ChronoQuark extends Quark {
-	@JsonProperty("timestampKey")
+	@JsonProperty("timestamps")
 	@DsonOutput(DsonOutput.Output.ALL)
-	private String timestampKey;
-
-	@JsonProperty("timestampValue")
-	@DsonOutput(DsonOutput.Output.ALL)
-	private long timestampValue;
+	private Map<String, Long> timestamps = new HashMap<>();
 
 	private ChronoQuark() {
 	}
 
-	public ChronoQuark(String timestampkey, long timestampValue) {
-		this.timestampKey = Objects.requireNonNull(timestampkey);
-		this.timestampValue = timestampValue;
+	public ChronoQuark(String timestampKey, long timestampValue) {
+		this.timestamps.put(Objects.requireNonNull(timestampKey, "timestampKey is required"), timestampValue);
+	}
+
+	public long getTimestamp() {
+		return getTimestamp(Timestamps.DEFAULT);
+	}
+
+	public long getTimestamp(String type) {
+		return this.timestamps.getOrDefault(Objects.requireNonNull(type, "type is required"), 0L);
 	}
 
-	public String getTimestampKey() {
-		return this.timestampKey;
+	public void setTimestamp(String type, long timestamp) {
+		this.timestamps.put(Objects.requireNonNull(type, "type is required"), timestamp);
```

Comment:
Again, would like to see prereqs separate from code body.

Reply:
Actually, can we get rid of this method by providing a richer set of constructors?

This is the only mutating method that I can see for the class, so objects would be immutable if this method was removed.